bundles/org.openhab.voice.actiontemplatehli/src/main/java/org/openhab/voice/actiontemplatehli/internal/ActionTemplateInterpreter.java

```diff
@@ -0,0 +1,1251 @@
+            if (response == null) {
+                logger.debug("silent mode; no response");
+                return "";
+            }
+            logger.debug("response: {}", response);
+            return response;
+        } catch (IOException e) {
+            logger.debug("IOException while interpreting: {}", e.getMessage(), e);
+            var message = e.getMessage();
+            throw new InterpretationException(message != null ? message : "Unknown error");
+        } catch (RuntimeException e) {
+            var message = e.getMessage();
+            logger.debug("RuntimeException while interpreting: {}", e.getMessage(), e);
+            throw new InterpretationException(message != null ? message : "Unknown error");
+        }
+    }
+
+    private @Nullable String processAction(String words, @Nullable NLPInterpretationResult result)
+            throws InterpretationException, IOException {
+        if (result != null) {
+            if (!result.actionConfig.read) {
+                return sendItemCommand(result.targetItem, words, result.actionConfig, result.placeholderValues);
+            } else {
+                return readItemState(result.targetItem, result.actionConfig);
+            }
+        } else {
+            throw new InterpretationException(config.unhandledMessage);
+        }
+    }
+
+    private NLPInfo getNLPInfo(String text) throws IOException {
+        logger.debug("Processing: '{}'", text);
+        var tokens = tokenizeText(text);
+        var tags = languagePOSTagging(tokens);
+        var lemmas = languageLemmatize(tokens, tags);
+        logger.debug("tokens: {}", List.of(tokens));
+        logger.debug("tags: {}", List.of(tags));
+        logger.debug("lemmas: {}", List.of(lemmas));
+        return new NLPInfo(tokens, lemmas, tags);
+    }
+
+    private @Nullable NLPInterpretationResult checkActionConfigs(String text, String[] tokens, String[] tags,
+            String[] lemmas) throws IOException {
+        // item defined actions have priority over type defined actions
+        var result = checkItemActions(text, tokens, tags, lemmas);
+        if (result != null) {
+            return result;
+        }
+        return checkTypeActionsConfigs(text, tokens, tags, lemmas);
+    }
+
+    private @Nullable NLPInterpretationResult checkItemActions(String text, String[] tokens, String[] tags,
+            String[] lemmas) throws IOException {
+        // Check item with action config
+        var itemsWithActions = getItemsWithActionConfigs();
+        Item targetItem = null;
+        ActionTemplateConfiguration targetActionConfig = null;
+        // store data to restore placeholder values
+        List<NLPPlaceholderData> placeholderValues = null;
+        // store span of dynamic placeholder, to invalidate others
+        Span dynamicSpan = null;
+        int matchScore = 0;
+        for (var entry : itemsWithActions.entrySet()) {
+            var actionConfigs = entry.getValue();
```

Comment:
This part applies to actions added to the item metadata so I think it makes no sense to check the items tags, as you are defining the action just for one item. On the other place it's checking actions linked by multiple items by its item type so I see the benefit on filter by tags there.
I just checked it, in case it was wrong but it's correctly indicated on the readme
```
requiredTags: allow to restrict the members targeted by tags when matching by type.
```

Reply:
Ok, makes sense.